.travis.yml

```diff
@@ -18,6 +18,17 @@ addons:
     - texlive-generic-recommended
     - texlive-xetex
     - latex-xcolor
+
+deploy:
+  provider: pages
+  skip-cleanup: true
+  github-token: $GITHUB_TOKEN  # Set in the settings page of your repository, as a secure variable
+  keep-history: true
+  local-dir: _site
+  verbose: true
+  on:
+    branch: master
```

Comment:
So we only push to `gh-pages` if we're building/testing `master`. Is this what we want? (or were the pages updated every time before?)

If we do want that, can we see the output of the build somewhere? The only thing I could find is https://docs.travis-ci.com/user/uploading-artifacts/, which copies the build artifacts to an s3 account.

Reply:
As it is this will only push to `gh-pages` when `master` is built. i.e., only when a PR is merged. Previously it was pushing all the time!! which I think it's dangerous.

I do agree that we can built in the PRs and push to artifacts. (in a different PR).